mobile/src/people/departments/departments-horizontal-scrollable-list.tsx

```diff
@@ -0,0 +1,111 @@
+    private employeeCards: EmployeeCardWithAvatar[];
+    private animatedValue: Animated.Value;
+    private buttonText: Text;
+    private currentPage: number;
+
+    public componentDidMount() {
+        // this.animate.bind(this, Easing.bounce);
+        if (this.props.departmentsTreeNodes !== null) {
+            this.props.requestEmployeesForDepartment(this.props.departmentsTreeNodes[0].departmentId);
+        }
+    }
+
+    public onMomentumScrollEnd(event: NativeSyntheticEvent<NativeScrollEvent>) {
+        var offset = event.nativeEvent.contentOffset;
+        if (offset) {
+            this.currentPage = Math.round(offset.x / Dimensions.get('window').width) + 1;
+            if (this.currentPage > this.employeeCards.length) {
+                this.props.requestEmployeesForDepartment(this.props.headDepartment.departmentId);
+                const stubDN: DepartmentsTreeNode = {
+                    departmentAbbreviation: null,
+                    departmentChiefId: this.props.headDepartment.head.employeeId,
+                    head: this.props.headDepartment.head,
+                    parent: null,
+                    children: null,
+                    subordinates: null,
+                    departmentId: 'subordinates'
+                };
+                this.props.updateDepartmentIdsTree(this.props.treeLevel, stubDN);
+            } else {
+                const visibleCard: EmployeeCardWithAvatar = this.employeeCards[this.currentPage - 1];
+                visibleCard.revealNeighboursAvatars(true);
+                const visibleDepartment = this.props.headDepartment.children[this.currentPage - 1];
+                this.props.updateDepartmentIdsTree(this.props.treeLevel, visibleDepartment);
+                this.props.requestEmployeesForDepartment(visibleDepartment.departmentId);
+                if (visibleDepartment.children !== null && visibleDepartment.children.length > 0) {
+                    this.props.requestEmployeesForDepartment(visibleDepartment.children[0].departmentId);
+                }
+            }
+        }
+    }
+
+    public onScrollBeginDrag(event: NativeSyntheticEvent<NativeScrollEvent>) {
+        this.employeeCards.forEach(card => {
+            card.revealNeighboursAvatars(false);
+        });
+    }
+
+    public render() {
+        this.employeeCards = [];
+        
+        const { headDepartment } = this.props;
+        const subDepartments = headDepartment != null && headDepartment.children != null ? headDepartment.children : null;
+        const subordinates = headDepartment != null && headDepartment.subordinates != null ? headDepartment.subordinates : null;
+
+        return <View>
+            <Animated.ScrollView 
+                horizontal 
+                pagingEnabled 
+                showsHorizontalScrollIndicator={false}
+                onMomentumScrollEnd={this.onMomentumScrollEnd.bind(this)}
```

Comment:
And again, don't use new lambdas, binds in the props, it every time returns new instances, causing re-render.

Use:

private onMomentumScrollEnd = (...) => {...}